db-service/lib/common/generic-pool.js

```diff
@@ -14,16 +14,17 @@ function _track_connections4(pool) {
     async acquire() {
       const connections = (this._trackedConnections ??= new Set())
       try {
-        this.dbc = await acquire.call(this)
+        let dbc = await acquire.call(this)
+        connections.add((dbc._beginStack = new Error('begin called from:')))
```

Comment:
This was intentionally done later. As the goal of the connection tracking is as follows:

```js
const connections = [
  cds.tx(),
  cds.tx(),
]
const [a,b] = await promise.all(connections)
// a will connect and fill the pool
// b will throw with the information that a is blocking the connection.
```
For clarity the tracked connections should only consist of successfully acquired connections and not all connections trying to acquire a connection. As they are not the root cause of the acquire timeout error.

Reply:
Is that relevant for SQLite only?